tools/run-tests.py

```diff
@@ -114,46 +114,49 @@ def create_binary(buildoptions):
 def run_jerry_tests():
     for job in jerry_tests_options:
         ret = create_binary(job.build_args)
+        if ret:
+            break
 
-        if not ret:
-            test_cmd = [TEST_RUNNER_SCRIPT, get_binary_path(job.out_dir), JERRY_TESTS_DIR]
-            if job.test_args:
-                test_cmd.extend(job.test_args)
+        test_cmd = [TEST_RUNNER_SCRIPT, get_binary_path(job.out_dir), JERRY_TESTS_DIR]
+        if job.test_args:
+            test_cmd.extend(job.test_args)
 
-            ret = run_check(test_cmd)
-        else:
+        ret = run_check(test_cmd)
```

Comment:
It's true, but if I only changed here, the next cycle of the loop will override this aggregated value if the build success, hiding the failing testrun.
And if we also aggregate the return value of ```create_binary()``` and also remove the related ```break``` statement, in case of an unsuccessful build the script would like to run a non-existing binary.

I've updated the patch, now the testrun breaks immediately when the build was unsuccessful, otherwise aggregate the return codes of test-steps.